### DISCUSSION TITLE
Up to date inventory levels without constantly re building?

### DISCUSSION THREAD

**Discussion by ghost** [gatsbyjs/gatsby#35939] [Help]
Hi there,

I've been testing a headless gatsby/shopify store using this plugin and I noticed that when I change stock levels on the shopify side, I have to rebuild gatsby in order for the stock level to update. 

Although Shopify handles this, I would quite like to prevent the user getting to checkout to find it's out of stock, and perhaps I could add another behaviour if this was the case. 

Does the plugin rely on a build to fetch data? Is there a way for it to stay up to date or will I have to use separate queries to get the latest availability?

Thank you

**Comment by ghost** [comment]
Ok I think I will solve by using fetch to get up to date data each time the page loads

**Comment by KyleAMathews** [comment]
How long do your builds take? Are they automated with webhooks? On Gatsby Cloud, the site should be updated in < 20s generally unless you have a larger store.

Fetching on the fly is great though if sales are very rapid so inventory can go out of stock while people are looking at pages.
